backend/src/schema/resolvers/reports.spec.js

```diff
@@ -287,3 +305,217 @@ describe('report', () => {
+      context: () => {
+        return {
+          driver,
+          user: authenticatedUser,
+        }
+      },
+    })
+    query = createTestClient(server).query
+    mutate = createTestClient(server).mutate
+  })
+
+  beforeEach(async () => {
+    authenticatedUser = null
+
+    moderator = await factory.create('User', {
+      id: 'mod1',
+      role: 'moderator',
+      email: 'moderator@example.org',
+      password: '1234',
+    })
+    user = await factory.create('User', {
+      id: 'user1',
+      role: 'user',
+      email: 'test@example.org',
+      password: '1234',
+    })
+    author = await factory.create('User', {
+      id: 'auth1',
+      role: 'user',
+      name: 'abusive-user',
+      email: 'abusive-user@example.org',
+    })
+    await instance.create('Category', {
+      id: 'cat9',
+      name: 'Democracy & Politics',
+      icon: 'university',
+    })
+
+    await Promise.all([
+      factory.create('Post', {
+        author,
+        id: 'p1',
+        categoryIds,
+        content: 'Interesting Knowledge',
+      }),
+      factory.create('Post', {
+        author: moderator,
+        id: 'p2',
+        categoryIds,
+        content: 'More things to do …',
+      }),
+      factory.create('Post', {
+        author: user,
+        id: 'p3',
+        categoryIds,
+        content: 'I am at school …',
+      }),
+    ])
+    await Promise.all([
+      factory.create('Comment', {
+        author: user,
+        id: 'c1',
+        postId: 'p1',
+      }),
+    ])
+
+    authenticatedUser = await user.toJson()
+    await Promise.all([
+      mutate({
+        mutation: reportMutation,
+        variables: {
+          resourceId: 'p1',
+          reasonCategory: 'other',
+          reasonDescription: 'This comment is bigoted',
+        },
+      }),
+      mutate({
+        mutation: reportMutation,
+        variables: {
+          resourceId: 'c1',
+          reasonCategory: 'discrimination_etc',
+          reasonDescription: 'This post is bigoted',
+        },
+      }),
+      mutate({
+        mutation: reportMutation,
+        variables: {
+          resourceId: 'auth1',
+          reasonCategory: 'doxing',
+          reasonDescription: 'This user is harassing me with bigoted remarks',
+        },
+      }),
+    ])
+    authenticatedUser = null
+  })
+
+  afterEach(async () => {
+    await factory.cleanDatabase()
+  })
+
+  describe('unauthenticated', () => {
+    it('throws authorization error', async () => {
+      authenticatedUser = null
+      expect(query({ query: reportsQuery })).resolves.toMatchObject({
+        data: { reports: null },
+        errors: [{ message: 'Not Authorised!' }],
+      })
+    })
+
+    it('role "user" gets no reports', async () => {
+      authenticatedUser = await user.toJson()
+      expect(query({ query: reportsQuery })).resolves.toMatchObject({
+        data: { reports: null },
+        errors: [{ message: 'Not Authorised!' }],
+      })
+    })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-14T08:23:50Z" title="Monday, October 14th 2019, 10:23:50 am +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -287,3 +291,227 @@ describe('report', () => {
     })
   })
 })
+
+describe('reports query', () => {
+  let query
+  let mutate
+  let authenticatedUser = null
+  let moderator
+  let user
+  let author
+  const categoryIds = ['cat9']
+
+  const reportMutation = gql`
+    mutation($resourceId: ID!, $reasonCategory: String!, $reasonDescription: String!) {
+      report(
+        resourceId: $resourceId
+        reasonCategory: $reasonCategory
+        reasonDescription: $reasonDescription
+      ) {
+        type
+      }
+    }
+  `
+  const reportsQuery = gql`
+    query {
+      reports(orderBy: createdAt_desc) {
+        createdAt
+        reasonCategory
+        reasonDescription
+        submitter {
+          id
+        }
+        resourceId
+        type
+        user {
+          id
+        }
+        post {
+          id
+        }
+        comment {
+          id
+        }
+      }
+    }
+  `
+
+  beforeAll(() => {
+    const { server } = createServer({
+      context: () => {
+        return {
+          driver,
+          user: authenticatedUser,
+        }
+      },
+    })
+    query = createTestClient(server).query
+    mutate = createTestClient(server).mutate
+  })
+
+  beforeEach(async () => {
+    authenticatedUser = null
+
+    moderator = await factory.create('User', {
+      id: 'mod1',
+      role: 'moderator',
+      email: 'moderator@example.org',
+      password: '1234',
+    })
+    user = await factory.create('User', {
+      id: 'user1',
+      role: 'user',
+      email: 'test@example.org',
+      password: '1234',
+    })
+    author = await factory.create('User', {
+      id: 'auth1',
+      role: 'user',
+      name: 'abusive-user',
+      email: 'abusive-user@example.org',
+    })
+    await instance.create('Category', {
+      id: 'cat9',
+      name: 'Democracy & Politics',
+      icon: 'university',
+    })
+
+    await Promise.all([
+      factory.create('Post', {
+        author,
+        id: 'p1',
+        categoryIds,
+        content: 'Not for you',
+      }),
+      factory.create('Post', {
+        author: moderator,
+        id: 'p2',
+        categoryIds,
+        content: 'Already seen post mention',
+      }),
+      factory.create('Post', {
+        author: user,
+        id: 'p3',
+        categoryIds,
+        content: 'You have been mentioned in a post',
+      }),
+    ])
+    await Promise.all([
+      factory.create('Comment', {
+        author: user,
+        id: 'c1',
+        postId: 'p1',
+      }),
+    ])
+
+    authenticatedUser = await user.toJson()
+    await Promise.all([
+      mutate({
+        mutation: reportMutation,
+        variables: {
+          resourceId: 'p1',
+          reasonCategory: 'other',
+          reasonDescription: 'This comment is bigoted',
+        },
+      }),
+      mutate({
+        mutation: reportMutation,
+        variables: {
+          resourceId: 'c1',
+          reasonCategory: 'discrimination-etc',
+          reasonDescription: 'This post is bigoted',
+        },
+      }),
+      mutate({
+        mutation: reportMutation,
+        variables: {
+          resourceId: 'auth1',
+          reasonCategory: 'doxing',
+          reasonDescription: 'This user is harassing me with bigoted remarks',
+        },
+      }),
+    ])
+    authenticatedUser = null
+  })
+
+  afterEach(async () => {
+    await factory.cleanDatabase()
+  })
+
+  describe('unauthenticated', () => {
+    it('throws authorization error', async () => {
+      authenticatedUser = null
+      const { data, errors } = await query({ query: reportsQuery })
+      expect(data).toEqual({
+        reports: null,
+      })
+      expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
+    })
+
+    it('roll "user" gets no reports', async () => {
+      authenticatedUser = await user.toJson()
+      const { data, errors } = await query({ query: reportsQuery })
+      expect(data).toEqual({
+        reports: null,
+      })
+      expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
+    })
```


please refactor this to use one `expect` and the new, agreed upon sytanx
see https://github.com/Human-Connection/Human-Connection/pull/1884/files#diff-a9111c79c1bda625b0005a6bf3ca91beR77

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-14T15:21:18Z" title="Monday, October 14th 2019, 5:21:18 pm +02:00">Oct 14, 2019</time>_

---


Thanks for the nice hint 😍, I have refactored yet locally …